apps/src/music/blockly/blocks/toolboxMode.js

```diff
@@ -0,0 +1,63 @@
+import {BlockTypes} from '../blockTypes';
+import {categoryTypeToLocalizedName, dynamicCategoryLabels} from '../toolbox';
+import {Category} from '../toolbox/types';
+
+const validStaticCategories = Object.entries(Category)
+  .filter(
+    ([key, value]) =>
+      // Functions and Variables are handled by dynamic categories
+      value !== Category.Functions && value !== Category.Variables
+  )
+  .map(([key, value]) => [categoryTypeToLocalizedName[value], value]);
+
+const validDynamicCategories = Object.entries(dynamicCategoryLabels).map(
+  ([key, value]) => [key, value]
+);
+
+/**
+ * Represents a dynamic category block definition for Blockly.
+ * This block allows levelbuilder to specify the start of a new category
+ * when in toolbox mode.
+ */
+export const staticCategoryBlock = {
+  definition: {
+    type: BlockTypes.CATEGORY,
+    message0: 'Category %1',
+    args0: [
+      {
+        type: 'field_dropdown',
+        name: 'CATEGORY',
+        options: validStaticCategories,
+      },
+    ],
+    inputsInline: true,
+    style: 'loop_blocks',
+    tooltip:
+      'Indicates the start of a new category. All blocks below this point will be contained in this category.',
+  },
+  generator: () => '\n',
+};
+
+/**
+ * Represents a dynamic category block definition for Blockly. This block allows
+ * levelbuilders to create auto-populated Functions or Variables categories when
+ * in toolbox mode.
+ */
+export const dynamicCategoryBlock = {
```

Comment:
Out of curiosity, are these names from Blockly? (dynamic v. static)
Seems like dynamic means it could be changed and static would they're not editable (like Functions and Variables).

Reply:
Yes, they're from Blockly. Static means the contents won't ever change (for the user), whereas dynamic category contents can change as the user defines new variables and functions.

Reply:
Gotcha - thanks for the explanation!